utilix/batchq.py

```diff
@@ -1,265 +1,300 @@
-        exclude_nodes = '#SBATCH --exclude={exclude_nodes}'.format(exclude_nodes=exclude_nodes)
-    else:
-        exclude_nodes = ''
-
-    if not dependency is None:
-        if isinstance(dependency, list):
-            # Handle list of strings
-            job_ids = ":".join(dependency)
-        elif isinstance(dependency, str):
-            # Handle single string
-            job_ids = dependency
-        else:
-            raise ValueError(f'dependency should be list or str but given {type(dependency)}')
-        
-        dependency = "--dependency=afterok:"+job_ids+" --kill-on-invalid-dep=yes"
-    
-    else:
-        dependency = ''
-
-
-
-    sbatch_script = sbatch_template.format(jobname=jobname, log=log, qos=qos, partition=partition,
-                                           account=account, job=jobstring, mem_per_cpu=mem_per_cpu,
-                                           cpus_per_task=cpus_per_task, hours=hours, node=node,
-                                           exclude_nodes=exclude_nodes)
-
-    if dry_run:
-        print("=== DRY RUN ===")
-        print(sbatch_script)
-        return
-
-    if sbatch_file is None:
-        remove_file = True
-        _, sbatch_file = tempfile.mkstemp(suffix='.sbatch')
-    else:
-        remove_file = False
-
-    with open(sbatch_file, 'w') as f:
-        f.write(sbatch_script)
-
-    command = "sbatch %s %s" % (dependency, sbatch_file)
-    if not sbatch_file:
-        print("Executing: %s" % command)
-
-    process = subprocess.Popen(shlex.split(command), stdout=subprocess.PIPE, stdin=subprocess.PIPE, stderr=subprocess.PIPE)
-    output, error = process.communicate()
- 
-
-    # Check if the job submission was successful
-    if process.returncode == 0:
-        # Extract the job ID from the output
-        job_id = output.decode().strip().split()[-1]
-    else:
-        job_id = None
-        # Handle the case where the job submission failed
-        print("Job submission failed:", error.decode())
-        
-    if remove_file:
-        os.remove(sbatch_file)
-
-    return job_id
-
-def count_jobs(string=''):
-    username = os.environ.get("USER")
-    output = subprocess.check_output(shlex.split("squeue -u %s" % username))
-    lines = output.decode('utf-8').split('\n')
-    return len([job for job in lines if string in job])
+            file_discriptor, exec_file = tempfile.mkstemp(suffix=".sh", dir=TMPDIR[self.partition])
+            _make_executable(exec_file)
+            os.write(file_discriptor, bytes("#!/bin/bash\n" + self.jobstring, "utf-8"))
+        bind_string = " ".join([f"--bind {b}" for b in self.bind])
+        image = os.path.join(SINGULARITY_DIR[self.partition], self.container)
+        if not os.path.exists(image):
+            raise FileNotFoundError(f"Singularity image {image} does not exist")
+        new_job_string = (
+            f"unset X509_CERT_DIR\n"
+            f'if [ "$INSTALL_CUTAX" == "1" ]; then unset CUTAX_LOCATION; fi\n'
```

Comment:
It's doing the same thing as the original code, but the printing message is missing here. Could you add something like #84 to print the information explicitly?